simulation_server.py

```diff
@@ -324,14 +325,20 @@ def runWebotsInThread(client):
                                     envVarDocker["THEIA_VOLUME"] = volume
                                     envVarDocker["THEIA_PORT"] = port + 500
                                     client.websocket.write_message('ide: enable')
-                                elif info[1].strip() == 'default_controller':
-                                    default_controller = info[2]
-                                    dockerComposePath = config['dockerConfDir'] + "/docker-compose-benchmark.yml"
-                                    envVarDocker["DEFAULT_CONTROLLER"] = default_controller
+                                elif info[1].strip() == 'competition':
+                                    client.competition = True
+                                    volume = info[2]
+                                    dockerComposePath = config['dockerConfDir'] + "/docker-compose-competition.yml"
+                                    envVarDocker["THEIA_VOLUME"] = volume
                                     envVarDocker["THEIA_PORT"] = port + 500
                                     client.websocket.write_message('ide: enable')
                                     # using hard link so that the COPY command in the Dockerfile will work on the launcher file
                                     os.system(f'ln {config["dockerConfDir"]}/remote_controller_launcher.py')
+                                    with open(world, 'r') as world_file:
+                                        world_content = world_file.read()
+                                    world_content = world_content.replace('controller "participant"', 'controller "<extern>"')
```

Comment:
Shouldn't setup things for the opponent as well?
```suggestion
                                    world_content = world_content.replace('controller "participant"', 'controller "<extern>"')
                                    world_content = world_content.replace('controller "opponent"', 'controller "<extern>"')
```

Reply:
This docker-compose is used for people to test the competition online against the example controller provided in the files of the competition. We just need to isolate the participant's controller inside its own container.